test/jdk/javax/swing/JFileChooser/FileChooserSymLinkTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/* @test
+   @bug 8281966
+   @key headful
+   @requires (os.family == "windows")
+   @library /java/awt/regtesthelpers
+   @build PassFailJFrame
+   @summary Test to check if the absolute path of Symbolic Link folder
+            is valid on ValueChanged property listener.
+   @run main/manual FileChooserSymLinkTest
+*/
+
+import java.awt.Dimension;
+import java.awt.Robot;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+public class FileChooserSymLinkTest{
+    static JFrame frame = null;
+    static PassFailJFrame passFailJFrame = null;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    initialize();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                } catch (InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+        robot.delay(2000);
+        robot.waitForIdle();
```

Comment:
Robot is not required. You can safely remove these two lines.

The following line just starts to wait for the user to press `Pass` or `Fail`.

Reply:
> It does not work.
> 
> In the single-selection case, I see the following output:
> 
> ```
> Absolute Path : null
> Absolute Path : C:\filechooser\link
> ```
> 
> Above, `null` corresponds to `C:\filechooser\target` folder which is the target of the `link`.
> 
> Neither does it work in the multi-selection case:
> 
> ```
> Absolute Path : C:\filechooser
> Absolute Path : null
> Absolute Path : C:\filechooser\target
> ```
> 
> But now it doesn't work in a different way: `link` is reported as `null`.
> 
> I can't see none of the comments addressed in the test even though you explicitly added comments that you updated the code. A missing push?

Yeah, sorry for that, I had missed to push the updates.

Reply:
> > It does not work.
> > In the single-selection case, I see the following output:
> > ```
> > Absolute Path : null
> > Absolute Path : C:\filechooser\link
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Above, `null` corresponds to `C:\filechooser\target` folder which is the target of the `link`.
> > Neither does it work in the multi-selection case:
> > ```
> > Absolute Path : C:\filechooser
> > Absolute Path : null
> > Absolute Path : C:\filechooser\target
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But now it doesn't work in a different way: `link` is reported as `null`.
> > I can't see none of the comments addressed in the test even though you explicitly added comments that you updated the code. A missing push?
> 
> Yeah, sorry for that, I had missed to push the updates.

For link it doesn't work, I have updated only for symbolic links and not for .lnk. .lnk logic is retained as it is.

Reply:
> In the single-selection case, I see the following output:
> 
> ```
> Absolute Path : null
> Absolute Path : C:\filechooser\link
> ```

Sorry about that. Upon re-testing, this scenario worked fine with your original changes.

> For link it doesn't work, I have updated only for symbolic links and not for .lnk. .lnk logic is retained as it is.

I didn't test `.lnk` files, I tested with symbolic links created using `mklink` command.

Your changes worked in single-selection case. Sorry, I must have done something wrong.